### ISSUE TITLE
'Warning: To load an ES module, set "type": "module" in the package.json or use the .mjs extension.' when running the default app

### ISSUE THREAD

**Issue by hankertrix** [expo/expo#30122]
### Summary

The warning 'Warning: To load an ES module, set "type": "module" in the package.json or use the .mjs extension.' appears twice when running the default application.

### What platform(s) does this occur on?

iOS

### SDK Version

51.0.14

### Environment

  expo-env-info 1.2.0 environment info:
    System:
      OS: Linux 6.9 EndeavourOS
      Shell: 5.2.26 - /bin/bash
    Binaries:
      Node: 22.3.0 - /usr/bin/node
      npm: 10.8.1 - /usr/bin/npm
    npmPackages:
      expo: ~51.0.14 => 51.0.17 
      expo-router: ~3.5.16 => 3.5.17 
      react: 18.2.0 => 18.2.0 
      react-dom: 18.2.0 => 18.2.0 
      react-native: 0.74.2 => 0.74.2 
      react-native-web: ~0.19.10 => 0.19.12 
    Expo Workflow: managed

### Minimal reproducible example

```sh
npx create-expo-app@latest test
cd test
npx expo start
```

**Comment by eilifjohansen** [comment]
I am getting the same warnings:

```
(node:50593) [DEP0040] DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.
(Use `node --trace-deprecation ...` to show where the warning was created)
Starting Metro Bundler
(node:50593) Warning: To load an ES module, set "type": "module" in the package.json or use the .mjs extension.
(node:50593) Warning: To load an ES module, set "type": "module" in the package.json or use the .mjs extension.
```
▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄

**Comment by byCedric** [comment]
Hi all! Expo supports both the "maintenance LTS" (Node 18) and the "active LTS" (Node 20) - [see Node releases](https://nodejs.org/en/about/previous-releases). That said, Node 22 runs fine, even with these warnings. This is on our radar and is caused by two separate systems.

- The Punycode deprecation warning should be fixed by #29511
- The `Warning: to load an ES module, ...` is caused by how we generate each project's Expo Router type definitions.

Because these two issues have been isolated and are on our radar, I'll close this issue. It will likely be resolved in SDK 52.

Hope this helps!
